_search-plugins/knn/approximate-knn.md

```diff
@@ -127,10 +127,25 @@ GET my-knn-index-1/_search
 }
 ```
 
-`k` is the number of neighbors the search of each graph will return. You must also include the `size` option, which
-indicates how many results the query actually returns. The plugin returns `k` amount of results for each shard
-(and each segment) and `size` amount of results for the entire query. The plugin supports a maximum `k` value of 10,000.
-Starting in OpenSearch 2.14, in addition to using the `k` variable, both the `min_score` and `max_distance` variables can be used for [radial search]({{site.url}}{{site.baseurl}}/search-plugins/knn/radial-search-knn/).
+### The number of returned results
+
+In the preceding query, `k` is the number of neighbors the search of each graph returns. You must also include the `size` option, indicating the final number of results you want the query to return.  
+
+For NMSLIB and Faiss engines, `k` is the maximum number of documents returned for all segments of a shard. For the Lucene engine, `k` is the number of documents returned for a shard. The maximum value of `k` is 10,000.
+
+For any engine, each shard returns `size` results to the coordinator node. Thus, the total number of results the coordinator node receives is `size * number of shards`. After the coordinator node consolidates the results received from all nodes, the query returns the top `size` results.
+
+The following table provides examples of the number of results returned by various engines in several scenarios.
```

Comment:
@kolchfa-aws why we are not adding the full table here?

Reply:
@navneet1v It is the full table, but each pair of two identical rows is one row that contains Faiss/NMSLIB and Lucene results side by side so it's easier for the users to compare those.

![Uploading Screenshot 2024-05-28 at 5.01.11 PM.png…]()

Reply:
oh.. make sense. So you added 1 more column. thanks

Reply:
Can we add 1 more line above this, saying that this the table assumes data there are enough number of documents across shards and segments.

Reply:
@navneet1v Done!